codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/PresignerGenerator.kt

```diff
@@ -0,0 +1,91 @@
+package software.amazon.smithy.aws.swift.codegen
+
+import software.amazon.smithy.aws.swift.codegen.AWSClientRuntimeTypes.Core.AWSClientConfiguration
+import software.amazon.smithy.aws.swift.codegen.middleware.AWSSigningMiddleware
+import software.amazon.smithy.aws.swift.codegen.model.traits.Presignable
+import software.amazon.smithy.model.knowledge.OperationIndex
+import software.amazon.smithy.model.shapes.OperationShape
+import software.amazon.smithy.model.shapes.ServiceShape
+import software.amazon.smithy.swift.codegen.ClientRuntimeTypes.Http.SdkHttpRequest
+import software.amazon.smithy.swift.codegen.ClientRuntimeTypes.Middleware.NoopHandler
+import software.amazon.smithy.swift.codegen.SwiftDelegator
+import software.amazon.smithy.swift.codegen.SwiftWriter
+import software.amazon.smithy.swift.codegen.core.CodegenContext
+import software.amazon.smithy.swift.codegen.core.toProtocolGenerationContext
+import software.amazon.smithy.swift.codegen.integration.MiddlewareExecutionGenerator
+import software.amazon.smithy.swift.codegen.integration.SwiftIntegration
+import software.amazon.smithy.swift.codegen.model.expectShape
+
+data class PresignableOperation(
+    val serviceId: String,
+    val operationId: String,
+    // TODO ~ Implementation of embedded presigned URLs is TBD
```

Comment:
question, what is even is embedded presigned URLs?

Reply:
I'm not sure either.  This was also a TODO in kotlin's code, so I carried it over.